app/components/alerts/AlertsList.js

```diff
@@ -5,46 +5,82 @@ import { RefreshControl, FlatList, ScrollView, View, Image, Text } from 'react-n
 import { ListItem } from 'react-native-elements';
 import AlertsStyles from '../../styles/AlertStyles';
 import emptyImage from '../../images/alerts.png';
-import { getAlertsList } from '../../actions';
+import { getAlertsList, updateUserLastAlertSeen } from '../../actions';
 
 class AlertsList extends React.Component {
-  constructor(props) {
-    super(props);
+	constructor(props) {
+		super(props);
 
-    this.state = {
+		props.currentAlerts.sort(function(a,b){
+			return (b.sentDate == null)-(a.sentDate == null) || b.sentDate - a.sentDate;
+		}).reverse();
+
+		this.state = {
+			userId: props.userId,
+			lastAlertSeen: props.lastAlertSeen,
 			getAlertsList: props.getAlertsList,
+			updateUserLastAlertSeen: props.updateUserLastAlertSeen,
 			currentAlerts: props.currentAlerts,
 			isRefreshing: false,
-    };
+		};
 
-    this.renderListItem = this.renderListItem.bind(this);
-    this.onRefresh = this.onRefresh.bind(this);
-  }
+    this.state.getAlertsList(); //ensure newest alerts shown w/o refreshing
+
+		this.renderListItem = this.renderListItem.bind(this);
+		this.onRefresh = this.onRefresh.bind(this);
+	}
 
-  componentWillReceiveProps(nextProps) {
+	componentWillReceiveProps(nextProps) {
 		// Avoiding refresh if possible
 		if (nextProps.currentAlerts !== this.state.currentAlerts) {
 			this.setState({ currentAlerts: nextProps.currentAlerts });
 		}
+		if (nextProps.lastAlertSeen !== this.state.lastAlertSeen) {
+			this.setState({ lastAlertSeen: nextProps.lastAlertSeen });
+		}
 		if (this.state.isRefreshing) this.setState({ isRefreshing: false });
   }
 
-  onRefresh() {
+		componentWillUnmount() {
+			const { userId, lastAlertSeen, currentAlerts } = this.state;
+			if(currentAlerts.length > 0){
+				this.state.updateUserLastAlertSeen(userId, lastAlertSeen, new Date(currentAlerts[0].sentDate));
```

Comment:
Oh actually, I think you're right.  When you're passing it back to the redux store, you're storing it as a date object (doesnt get converted to a string).  So, this might work.